### ISSUE TITLE
[BUG] Conflict with thermal expansion arcane ensorcellator

### ISSUE THREAD

**Issue by DemonfangArun** [ShetiPhian/Localizations-and-Issues#201]
Forge: 2800
Thermal Expansion: 5.5.2.39
EnderTanks 1.6.7

https://paste.dimdev.org/lepurozuhu.mccrash

**Comment by ShetiPhian** [comment]
Don't rely have anything to go on.
I'm not in the crash log and my mods do not add an enchant, or register to any Thermal Expansion machine.


I just created an instance with the same mods and can load into a world an play just fine.

**Comment by DemonfangArun** [comment]
That's what I thought, but something changed between this version and the last one that causes the machine to break. I put it into the following test environment and tested numerous times toggling EnderTanks on and off.

![screenshot001184](https://user-images.githubusercontent.com/17183816/51433704-8c4d8d00-1c1e-11e9-8564-816586df617d.png)

**Comment by ShetiPhian** [comment]
Was able to look at this again:

So after noticing I overlooked VanillaFix, and installing it, I started getting its "crash" notice.
_off on a tangent: I wonder what its watching, as the problem is hidden without it. It would have appeared the machine just didn't work._

Thinking on it; `java.util.concurrent.ExecutionException`
Sounds like something is not behaving in a thread safe manner

> but something changed between this version and the last one

Points at my new Comparator updating code

So I wrapped my updater code into the task scheduler, to make sure its always on the correct thread
(as its triggered when the tanks inventory updates, which may not be on the correct thread)

I'm no longer getting the VanillaFix notice, and JEI shows recipes for the enchanting machine.
Here is a build to test: http://www.mediafire.com/file/hoj5dchhwgecejj/endertanks-1.12.0-1.6.8.jar/file
If it fixes it for you, I'll push it to Curse.
